### PR TITLE
[Bug] - DateInput icon issue

### PR DESCRIPTION
**Pull request by g4rry420**
Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>

#### What does this PR do?
Resolves the two icons in the dateinput

#### Where should the reviewer start?
components => DateInput.js

#### What testing has been done on this PR?
None

#### How should this be manually tested?
Using the code in the codesandbox of the #5889 

#### Do Jest tests follow these best practices?

N/A

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
N/A

#### What are the relevant issues?
#5889

#### Do the grommet docs need to be updated?
No

#### Is this change backwards compatible or is it a breaking change?
Backwards compatibe

### CODE REVIEW

**Comment by jcfilben** [comment]
Implementation looks good!

I am debating if we should have an `icon` prop for DateInput. I realize we need to allow setting the icon through `inputProps` to maintain backwards compatibility. I am wondering if we should mark setting the icon through `inputProps` as deprecated and encourage changing the DateInput through an `icon` prop since the icon is no longer within the input that inputProps is referring to. Curious to hear what you think @ericsoderberghp or if anyone else has thoughts on this

**Comment by g4rry420** [comment]
Good idea, I will make the changes

File: src/js/components/DateInput/DateInput.js [hunk 1]
```diff
@@ -185,7 +188,7 @@ const DateInput = forwardRef(
           id={id}
           dropProps={{ align: { top: 'bottom', left: 'left' }, ...dropProps }}
           dropContent={calendar}
-          icon={<CalendarIcon size={iconSize} />}
+          icon={icon || <CalendarIcon size={iconSize} />}
```
Comment (jcfilben) [inline src/js/components/DateInput/DateInput.js:191]: We should still allow the icon to be set through input props.
```suggestion
          icon={icon || MaskedInputIcon || <CalendarIcon size={iconSize} />}
```

File: src/js/components/DateInput/DateInput.js [hunk 2]
```diff
@@ -247,7 +250,7 @@ const DateInput = forwardRef(
             <Button
               onClick={open ? closeCalendar : openCalendar}
               plain
-              icon={<CalendarIcon size={iconSize} />}
+              icon={icon || <CalendarIcon size={iconSize} />}
```
Comment (jcfilben) [inline src/js/components/DateInput/DateInput.js:253]:
```suggestion
              icon={icon || MaskedInputIcon || <CalendarIcon size={iconSize} />}
```

File: src/js/components/DateInput/DateInput.js [hunk 3]
```diff
@@ -97,6 +98,8 @@ const DateInput = forwardRef(
       schema ? valueToText(normalizedDate, schema) : undefined,
     );
 
+    const { icon: MaskedInputIcon, ...restOfInputProps } = inputProps || {};
+
```
Comment (jcfilben) [inline src/js/components/DateInput/DateInput.js:102]: We should add a comment that changing the icon with inputProps is deprecated and also put a warning in the console noting this as well.
```suggestion
    // Setting the icon through `inputProps` is deprecated.
    // The `icon` prop should be used instead.
    const { icon: MaskedInputIcon, ...restOfInputProps } = inputProps || {};
    if (MaskedInputIcon) {
      console.warn(
        `Customizing the DateInput icon through inputProps is deprecated. 
Use the icon prop instead.`,
      );
    }

```

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**